scanner/src/main/kotlin/ScanController.kt

```diff
@@ -88,8 +95,20 @@ class ScanController(
      */
     private val scanResults = mutableMapOf<ScannerWrapper, MutableMap<KnownProvenance, MutableList<ScanResult>>>()
 
-    fun addProvenanceResolutionIssue(id: Identifier, issue: Issue) {
-        provenanceResolutionIssues.getOrPut(id) { mutableListOf() } += issue
+    /**
+     * Set the [issue] which failed package provenance resolution for the package denoted by [id].
+     */
+    fun putPackageProvenanceResolutionIssue(id: Identifier, issue: Issue) {
+        packageProvenanceResolutionIssues[id] = issue
+    }
+
+    /**
+     * Set the [issue] which failed nested provenance resolution for the given [provenance].
+     */
+    fun putNestedProvenanceResolutionIssue(provenance: KnownProvenance, issue: Issue) {
+        provenance.requireEmptyVcsPath()
```

Comment:
Is this check really required here? It seems a bit over the top to me, otherwise several other function should have to do similar checks. For example, should `putPackageProvenanceResolutionIssue` and `addIssue` should verify that the `id` does exist?
But if you think it is required, it would probably make more sense to verify that `provenance` is in the values of `packageProvenancesWithoutVcsPath`.